File: apis/src/main/java/com/beat/apis/ticket/application/dto/TicketUpdateRequest.java
package com.beat.apis.ticket.application.dto;

import java.time.LocalDateTime;
import java.util.List;

public record TicketUpdateRequest(
	Long performanceId,
	String performanceTitle,
	int totalScheduleCount,
	List<TicketUpdateDetail> bookingList
) {
	public static TicketUpdateRequest of(
		Long performanceId,
		String performanceTitle,
		int totalScheduleCount,
		List<TicketUpdateDetail> bookingList) {
		return new TicketUpdateRequest(performanceId, performanceTitle, totalScheduleCount, bookingList);
	}
}

File: apis/src/main/kotlin/com/beat/apis/ticket/api/request/TicketUpdateRequest.kt
package com.beat.apis.ticket.api.request

import jakarta.validation.Valid

data class TicketUpdateRequest(
    val performanceId: Long,
    val performanceTitle: String?,
    val totalScheduleCount: Int?,
    @field:Valid val bookingList: List<@Valid TicketUpdateDetail>,
)
